src/main/java/bio/terra/service/resourcemanagement/ResourceService.java

```diff
@@ -73,13 +81,30 @@ public GoogleBucketResource getOrCreateBucketForFile(Dataset dataset,
             billingProfile,
             null);
 
-        return bucketService.getOrCreateBucket(
-            dataLocationSelector.bucketForFile(datasetName, billingProfile),
-            projectResource,
+        List<UUID> bucketsForDataset = datasetBucketDao.getBucketForDatasetId(UUID.fromString(datasetId));
+        String bucketName = dataLocationSelector.bucketForFile(datasetId, billingProfile);
+
+        List<GoogleBucketResource> bucketsForBillingProfile = bucketsForDataset.stream()
+            .map(bucketUUID -> lookupBucket(bucketUUID.toString()))
+            .filter(bucket -> bucketIsForBillingProfile(bucket, billingProfile))
+        .collect(Collectors.toList());
+
+        if (bucketsForBillingProfile.size() > 0) {
+            GoogleBucketResource bucket = bucketsForBillingProfile.get(0);
```

Comment:
For the case where there's more than one, does it matter which one is picked?

Reply:
ideally, there shouldn't be more than one bucket. i think i asked the team if i should throw an error if we have multiple and we decided not to, but i can't remember the reasoning behind it -- will re-address

Reply:
Ok, adding a comment about the reasoning would be fine IMO.

Reply:
Ok, adding a comment about the reasoning would be fine IMO.

Reply:
Could we maybe log a warning if there's more than one?

Reply:
Can you add the log message

Reply:
oh wait nevermind...got lost in the messages